spellbook/models/x2y2/ethereum/x2y2_ethereum_events.sql

```diff
@@ -0,0 +1,180 @@
+    FROM  {{ source('x2y2_ethereum','X2Y2_r1_evt_EvProfit') }} prof 
+    INNER JOIN {{ source('x2y2_ethereum','X2Y2_r1_evt_EvInventory') }} inv ON inv.itemHash=prof.itemHash
+    LEFT JOIN {{ ref('tokens_ethereum_nft') }} tokens ON ('0x' || substring(get_json_object(inv.item, '$.data'), 155, 40)) = tokens.contract_address
+    WHERE taker NOT IN (SELECT contract_address FROM {{ ref('nft_ethereum_aggregators') }})
+    )
+
+, aggregator_routed_x2y2_txs_formatted AS (
+    SELECT block_time
+    , e721.to AS buyer
+    , seller
+    , token_id
+    , amount_raw
+    , currency_contract
+    , project_contract_address
+    , collection
+    , 'Bundle Trade' AS trade_type
+    , aggregator_name
+    , aggregator_address
+    , tx_hash
+    , txs.evt_index
+    , platform_fee_amount_raw
+    , platform_fee_percentage
+    , royalty_fee_amount_raw
+    , royalty_fee_percentage
+    , royalty_fee_receive_address
+    FROM aggregator_routed_x2y2_txs txs
+    LEFT JOIN {{ source('erc721_ethereum','evt_transfer') }} e721 ON txs.tx_hash = e721.evt_tx_hash
+        AND txs.token_id = e721.tokenId
+        AND e721.contract_address = txs.project_contract_address
+        AND to NOT IN (SELECT contract_address FROM {{ ref('nft_ethereum_aggregators') }})
+   )
+
+, direct_x2y2_txs_formated AS (
+    SELECT block_time
+    , buyer
+    , seller
+    , token_id
+    , amount_raw
+    , currency_contract
+    , project_contract_address
+    , collection
+    , 'Single Item Trade' AS trade_type
+    , aggregator_name
+    , aggregator_address
+    , tx_hash
+    , evt_index
+    , platform_fee_amount_raw
+    , platform_fee_percentage
+    , royalty_fee_amount_raw
+    , royalty_fee_percentage
+    , royalty_fee_receive_address
+    FROM direct_x2y2_txs
+    )
+
+, all_x2y2_txs AS (
+    SELECT * FROM aggregator_routed_x2y2_txs_formatted
+    UNION
+    SELECT * FROM direct_x2y2_txs_formated
+    )
+
+SELECT 'ethereum' AS blockchain
+, 'X2Y2' AS project
+, 'v1' AS version
+, txs.block_time
+, txs.token_id
+, txs.collection
+, CASE WHEN currency_contract='0x0000000000000000000000000000000000000000' THEN pu.price*txs.amount_raw/POWER(10, 18)
+    ELSE pu.price*txs.amount_raw/POWER(10, pu.decimals)
+    END AS amount_usd
+, CASE WHEN erct.evt_block_time IS NOT NULL THEN 'erc721'
+    ELSE 'erc1155'
+    END AS token_standard
+, trade_type
+, 1 AS number_of_items
+, CASE WHEN et.`from`=seller THEN 'Offer Accepted'
+    ELSE 'Buy'
+    END AS trade_category
+, 'Trade' AS evt_type
+, txs.seller
+, txs.buyer
+, CASE WHEN currency_contract='0x0000000000000000000000000000000000000000' THEN txs.amount_raw/POWER(10, 18)
+    ELSE txs.amount_raw/POWER(10, pu.decimals)
+    END AS amount_original
+, txs.amount_raw
+, CASE WHEN currency_contract='0x0000000000000000000000000000000000000000' THEN 'ETH'
+    ELSE pu.symbol
+    END AS currency_symbol
+, txs.currency_contract
+, txs.project_contract_address
+, aggregator_name
+, aggregator_address
+, txs.tx_hash
```

Comment:
Sure I can add it but I copied the code from opensea and looksrare here, so should I add block_number to all of those files as well? Because its missing everywhere

Reply:
No all good, I fixed this problem in other files in another PR just merged: https://github.com/duneanalytics/abstractions/pull/1291?no-redirect=1